packages/xpetra/src/CrsMatrix/Xpetra_TpetraBlockCrsMatrix_def.hpp

```diff
@@ -486,41 +486,66 @@ namespace Xpetra {
+    {
+      XPETRA_MONITOR("TpetraBlockCrsMatrix::getLocalRowView");
+      typename Tpetra::CrsGraph<LocalOrdinal,GlobalOrdinal,Node>::local_inds_host_view_type indices;
+      typename Tpetra::CrsMatrix<Scalar,LocalOrdinal,GlobalOrdinal,Node>::values_host_view_type values;
+
+      mtx_->getLocalRowView(LocalRow, indices, values);
+      Indices = ArrayView<const LocalOrdinal> (indices.data(), indices.extent(0));
+      Values = ArrayView<const Scalar> (values.data(), values.extent(0));
     }
 
 
     template<class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
     void 
     TpetraBlockCrsMatrix<Scalar,LocalOrdinal,GlobalOrdinal,Node>::
     getGlobalRowView(GlobalOrdinal GlobalRow, 
-                     ArrayView< const GlobalOrdinal > &indices, 
-                     ArrayView< const Scalar > &values) const
+                     ArrayView< const GlobalOrdinal > &Indices,
+                     ArrayView< const Scalar > &Values) const
     { 
         XPETRA_MONITOR("TpetraBlockCrsMatrix::getGlobalRowView"); 
+        typename Tpetra::CrsGraph<LocalOrdinal,GlobalOrdinal,Node>::global_inds_host_view_type indices;
+        typename Tpetra::CrsMatrix<Scalar,LocalOrdinal,GlobalOrdinal,Node>::values_host_view_type values;
+
         mtx_->getGlobalRowView(GlobalRow, indices, values);
+        Indices = ArrayView<const GlobalOrdinal> (indices.data(), indices.extent(0));
+        Values = ArrayView<const Scalar> (values.data(), values.extent(0));
     }
 
 
     template<class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
     void 
     TpetraBlockCrsMatrix<Scalar,LocalOrdinal,GlobalOrdinal,Node>::
     getGlobalRowCopy(GlobalOrdinal GlobalRow, 
-                     const ArrayView< GlobalOrdinal > &indices, 
-                     const ArrayView< Scalar > &values, 
-                     size_t &numEntries) const
+                     const ArrayView< GlobalOrdinal > &Indices,
+                     const ArrayView< Scalar > &Values,
+                     size_t &NumEntries) const
     { 
         XPETRA_MONITOR("TpetraBlockCrsMatrix::getGlobalRowCopy"); 
-        mtx_->getGlobalRowCopy(GlobalRow, indices, values, numEntries); 
-    }
-
+        typename Tpetra::CrsGraph<LocalOrdinal,GlobalOrdinal,Node>::nonconst_global_inds_host_view_type indices("indices",Indices.size());
+        typename Tpetra::CrsMatrix<Scalar,LocalOrdinal,GlobalOrdinal,Node>::nonconst_values_host_view_type values("values", Values.size());
 
-    template<class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
-    void 
-    TpetraBlockCrsMatrix<Scalar,LocalOrdinal,GlobalOrdinal,Node>::
-    getLocalRowView(LocalOrdinal LocalRow, ArrayView< const LocalOrdinal > &indices, 
-                    ArrayView< const Scalar > &values) const
-    { XPETRA_MONITOR("TpetraBlockCrsMatrix::getLocalRowView"); mtx_->getLocalRowView(LocalRow, indices, values); }
+        mtx_->getGlobalRowCopy(GlobalRow, indices, values, NumEntries);
```

Comment:
Is there a defined meaning for the underscore in the Trilinos naming scheme?  Is this used in order to name member variables (and therefore remove the need to use the `this` pointer)?

Reply:
Trilinos packages have a mixed set of standards. every class means a different code standard, but typically var_ is a member var, but not always, _t is also almost always a type

Reply:
Okay noted. The latter is _almost_ part of the standard; so it's good practice to self-enforce it. 

Regarding the latter, I favor using the `this` pointer rather than non-standardized underscore-based mangling. But if it's not a Trilinos requirement I will just stop being pedantic right here.

Reply:
this pointer can get overly verbose.  
Matts style is MyClass/myFunc/my_var_/my_arg
if you're uniform then you can really understand what the code is doing.  

You want the code readable but you want it to fit on a screen and a line so there's a balance.